src/hooks/routing/types.ts

```diff
@@ -1,10 +1,19 @@
 export enum RouterPreference {
-  PRICE,
-  SKIP,
   API,
   CLIENT,
 }
 
+export enum QuoteType {
+  TRADE = 'trade',
+  PRICE = 'price',
+  SKIP = 'skip',
+}
+
+export type QuoteConfig =
+  | { type: QuoteType.TRADE; preference: RouterPreference }
+  | { type: QuoteType.PRICE; preference: RouterPreference }
+  | { type: QuoteType.SKIP; preference?: undefined }
```

Comment:
Can this just be `| { type: QuoteType.SKIP }`? Why include the (double-)undefined `preference`?